server/src/main/java/org/apache/druid/server/security/AuthorizationResult.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server.security;
+
+import com.google.common.collect.ImmutableMap;
+import org.apache.druid.query.filter.DimFilter;
+import org.apache.druid.query.filter.TrueDimFilter;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+public class AuthorizationResult
```

Comment:
i know we discussed this offline, but on a second thought i kinda like the `getPermissionErrorMessage` method because:
- if enforces caller to send a policyRestrictionsNotPermitted param, which tells whether they cares about policy or not, which i think achieves that the caller can say they need all access no policy. 
- it returns an error message, which caller usually returns a forbidden exception with the message so it seems natural. 

i've set `policyRestrictionsNotPermitted` to true in all places except native sql path (i.e. QueryLifeCycle, AbstractStatement, AuthorizationUtils). to make if fail-safe, i added additional checks in MSQ and DART QueryMaker. Both now don't allow policy restrictions. 

if we were to use an enum, the caller needs to check if it's `ALLOW`, if not, call to get the error message or use a default message, which is one more step than `getPermissionErrorMessage`. WDYT?